compose/ui/ui-test-junit4/src/desktopTest/kotlin/androidx/compose/ui/test/BasicTestTest.kt

```diff
@@ -78,4 +77,25 @@ class BasicTestTest {
         val clockAfter = rule.mainClock.currentTime
         assertTrue(clockAfter > clockBefore, "performClick did not advance the test clock")
     }
+
+    @Test
+    fun advancingClockRunsRecomposition() {
+        rule.mainClock.autoAdvance = false
```

Comment:
Good point, I'll check what's the recommended way to do it in Android.

Reply:
It seems that JUnit is creating a new "rule" for each test.

Reply:
You're right. I didn't know that.

> JUnit creates a new instance of the test class for each test method that is run. Because each test rule is usually an instance variable of the test class, this means a new instance of the test rule is also created for each test method.
> 
> Test rules in JUnit are a way to define code that should be run before and after each test (similar to `@Before` and `@After` methods), but with more flexibility and reusability. 

I remember some tests where rule.mainClock was reset in After or Before. That's why I had a wrong impression.